### PR TITLE
Update from production/stage solr configs

### PR DESCRIPTION
**Pull request by atz**
These might not be "more correct" in terms of development direction, but
they are certainly more representative of what is running in production.
Please review and determine whether production is missing configuration
deployment or this config is lagging the real world.

### CODE REVIEW

File: solr_conf_4_testing/schema.xml [hunk 1]
```diff
@@ -17,7 +17,8 @@
     <field name="all_unstem_search" type="textNoStem" indexed="true" stored="true" multiValued="true" />
     <field name="vern_all_search" type="text" indexed="true" stored="true" multiValued="true" />
 
-    <!-- Format Field is now format_main_ssim: facet and display -->
+    <!-- Format Field: facet and display -->
+    <field name="format" type="string" indexed="true" stored="true" multiValued="true" />
```
Comment (ndushay) [inline solr_conf_4_testing/schema.xml:21]: "format" was the old searchworks field;  searchworks now uses format_main_ssim;  "format" should be sunsetted in spotlight.  (assuming spotlight still wants to be like searchworks)

File: solr_conf_4_testing/schema.xml [hunk 2]
```diff
@@ -144,12 +145,13 @@
     <field name="pub_country" type="text" indexed="true" stored="true" omitNorms="true"/>
     <!-- TODO: should pub_date_search be a date or a text field? -->
     <field name="pub_date_search" type="text" indexed="true" stored="true" omitNorms="true"/>
-    <!-- pub_date_sort now deprecated; replaced by pub_year_isi -->
```
Comment (ndushay) [inline solr_conf_4_testing/schema.xml:147]: spotlight should switch to pub_year_isi for sorting (if it wants to move the way SearchWorks should/did move).  It is an improved implementation in stanford-mods.  (though spotlight could be calling the right stanford-mods methods for indexing ... that is out of scope here.  Again, the comment shows the spirit of the direction we moved stanford-mods.

File: solr_conf_4_testing/schema.xml [hunk 3]
```diff
@@ -144,12 +145,13 @@
...
     <field name="pub_date_sort" type="alphaSort" indexed="true" stored="true" />
     <!-- Pub Date Facet Fields -->
-    <!-- pub_year_tisim is for date slider;  pub_year_w_approx_isi/pub_year_no_approx_isi are other facets -->
+    <!-- pub_date was facet and display, now deprecated -->
+    <field name="pub_date" type="string" indexed="true" stored="true" />
```
Comment (ndushay) [inline solr_conf_4_testing/schema.xml:151]: pub_date is the old way;  per the old comment, there are now other fields that are better for date slider and for facets.  pub_date was originally implemented to be everything in a date field, but that turned out to be not such a great plan.   I think I may have changed spotlight code to not use this field any more.

File: solr_conf_4_testing/schema.xml [hunk 4]
```diff
@@ -144,12 +145,13 @@
...
     <field name="pub_year_tisim" type="tint" indexed="true" stored="true" multiValued="true" />
     <field name="pub_display" type="string" indexed="false" stored="true" multiValued="true"/>
+    <field name="pub_date_display" type="string" indexed="false" stored="true"/>
```
Comment (ndushay) [inline solr_conf_4_testing/schema.xml:154]: it may be that spotlight no longer displays a field "pub_date_display" - I think I was removing unused cruft here.

File: solr_conf_4_testing/schema.xml [hunk 5]
```diff
@@ -259,7 +261,7 @@
 
   <!-- copy fields -->
   <copyField source="collection" dest="collection_search" />
-  <copyField source="pub_year_w_approx_isi" dest="pub_date_search" />
+  <copyField source="pub_date" dest="pub_date_search" />
```
Comment (ndushay) [inline solr_conf_4_testing/schema.xml:264]: ditto above with pub_date field comments  - I think it has been replaced?

**Comment by atz** [comment]
So the direction here is that stage and production are out of date, so this PR should be rejected, or rather reversed and applied to sul-solr-configs.

**Comment by ndushay** [comment]
So IIRC, we were updating the stanford-mods date parsing in the last SearchWorks sprint to accommodate a bunch of nasty dates, like B.C. dates and so on.  This triggered me to notice that the date logic in the stanford-mods gem was pretty wonky, sometimes preferring a date encoding type, and never paying attention to the keyDate attribute, which is the way to say "this one!" in MODS, for example. 

So I did a big update to the stanford-mods gem date parsing, and brought the new code into SearchWorks, and intended to bring the new code into Spotlight.  But I probably didn't get all the way finished with the Spotlight work -- and to keep everything working smoothly, I probably had indexed some of the stuff both the old way and the new way so we could seamlessly update the index and then the Spotlight exhibits to use the new index fields, and then remove the old index fields from the spotlight indexes.

So that's the background.  I think the schema file as it is without the PR is more the intent, but I don't know if exhibits production is ready for these schema files yet - you may need to reindex collections to ensure that there are no values in the Solr fields to be eliminated.

HTH.
